crates/sui-indexer/src/config.rs

```diff
@@ -354,4 +447,81 @@ mod test {
         // fullnode rpc url must be present
         parse_args::<JsonRpcConfig>([]).unwrap_err();
     }
+
+    #[test]
+    fn test_valid_pruning_config_file() {
+        let toml_str = r#"
+        epochs_to_keep = 5
+
+        [overrides]
+        objects_history = 10
+        transactions = 20
+        "#;
+
+        let opts = toml::de::from_str::<RetentionPolicies>(toml_str).unwrap();
+        assert_eq!(opts.epochs_to_keep, 5);
+        assert_eq!(opts.policies.get(&PrunableTable::ObjectsHistory), Some(&10));
+        assert_eq!(opts.policies.get(&PrunableTable::Transactions), Some(&20));
+        assert_eq!(opts.policies.len(), 2);
+    }
+
+    #[test]
+    fn test_pruning_options_from_file() {
+        let mut temp_file = NamedTempFile::new().unwrap();
+        let toml_content = r#"
+        epochs_to_keep = 5
+
+        [overrides]
+        objects_history = 10
+        transactions = 20
+        "#;
+        temp_file.write_all(toml_content.as_bytes()).unwrap();
+        let temp_path: PathBuf = temp_file.path().to_path_buf();
+        let pruning_options = PruningOptions {
+            pruning_config_path: Some(temp_path.clone()),
+        };
+        let mut retention_policies = pruning_options.load_from_file().unwrap();
+
+        // Assert the parsed values
+        assert_eq!(retention_policies.epochs_to_keep, 5);
+        assert_eq!(
+            retention_policies.get(&PrunableTable::ObjectsHistory),
+            Some(10)
+        );
+        assert_eq!(
+            retention_policies.get(&PrunableTable::Transactions),
+            Some(20)
+        );
+        assert_eq!(retention_policies.policies.len(), 2);
+
+        retention_policies = retention_policies.finalize();
+
```

Comment:
How about a test checking that other prunable tables are correctly covered by the default value in the finalized policies?

Reply:
added!